### PR TITLE
[CIS-251] Channel update

### PR DESCRIPTION
**Pull request by mantergo**
**In this PR:**

- Add `updateChannel` endpoint
- Add `updateChannel` functionality to `ChannelUpdater`
- Expose `updateChannel` functionality to `ChannelController`
- Make it possible to change channel name in `V3SampleApp`

### CODE REVIEW

File: StreamChatClient_v3/Workers/ChannelUpdater.swift [hunk 1]
```diff
@@ -26,6 +26,19 @@ class ChannelUpdater<ExtraData: ExtraDataTypes>: Worker {
...
+    /// Updates specific channel with new data.
+    /// - Parameters:
+    ///   - channelPayload: New channel data..
+    ///   - completion: Called when the API call is finished. Called with `Error` if the remote update fails.
+    func updateChannel(channelPayload: ChannelEditDetailPayload<ExtraData>, completion: ((Error?) -> Void)? = nil) {
+        apiClient.request(endpoint: .updateChannel(channelPayload: channelPayload)) {
+            switch $0 {
+            case .success: completion?(nil)
+            case let .failure(error): completion?(error)
+            }
```
Comment (buh) [inline StreamChatClient_v3/Workers/ChannelUpdater.swift:38]:
```suggestion
completion?($0.error)
```

Comment (mantergo) [reply StreamChatClient_v3/Workers/ChannelUpdater.swift:38]: Yep, was waiting for approve of https://github.com/GetStream/stream-chat-swift/pull/383. 
Will merge https://github.com/GetStream/stream-chat-swift/pull/383 and then will make this change.

File: StreamChatClient_v3/APIClient/Endpoints/ChannelEndpoints_Tests.swift [hunk 2]
```diff
@@ -136,3 +152,13 @@ final class ChannelEndpoints_Tests: XCTestCase {
         XCTAssertEqual(AnyEndpoint(expectedEndpoint), AnyEndpoint(endpoint))
     }
 }
+
+extension ChannelEditDetailPayload where ExtraData == DefaultDataTypes {
+    static var unique: Self {
```
Comment (VojtaStavik) [inline StreamChatClient_v3/APIClient/Endpoints/ChannelEndpoints_Tests.swift:157]: Interesting 👍

File: StreamChatClient_v3/Controllers/ChannelController.swift [hunk 3]
```diff
@@ -184,6 +184,29 @@ public class ChannelControllerGeneric<ExtraData: ExtraDataTypes>: Controller, De
...
 public extension ChannelControllerGeneric {
+    /// Updated channel with new data
+    /// - Parameters:
+    ///   - team: New team.
+    ///   - members: New members.
+    ///   - invites: New invites.
+    ///   - extraData: New `ExtraData`.
+    ///   - completion: The completion. Will be called on a **callbackQueue** when the network request is finished. If request fails, the completion
+    /// will be called with an error.
+    func updateChannel(team: String?,
```
Comment (VojtaStavik) [inline StreamChatClient_v3/Controllers/ChannelController.swift:195]: This method is not tested.

**Pull request merged**